### ISSUE TITLE
Does max_pooling3d support strides that are not same along height, width and depth?

### ISSUE THREAD

**Issue by nrahul** [tensorflow/tensorflow#2741]
I am using tensorflow to train a network that has 3D convolutions and 3D max-pooling layers. Some of the 3D max-pooling layers use a stride of 1 along the depth and strides of 2 along height and depth. This seems to be causing issues.

For example: The following script should print a tensor with values of '82' but somehow some of the values turn out to be -3.40282347e+38 

**import tensorflow as tf

sess = tf.InteractiveSession()
X = tf.Variable(tf.ones([8,16,112,112,3]))
W = tf.Variable(tf.ones([3,3,3,3,64]))
b = tf.Variable(tf.ones([64]))
out = tf.nn.relu(tf.nn.bias_add(tf.nn.conv3d(X, W, strides=[1,1,2,2,1], padding='VALID'),b))
out1 = tf.nn.max_pool3d(out, [1,2,3,3,1], strides=[1,1,2,2,1], padding='VALID')
sess.run(tf.initialize_all_variables())

print out1.eval()**

Additional notes:

- I looked into the tensorflow/python/kernel_tests/pooling_ops_3d_test.py file and all the test cases seem to have the same stride across all dimensions.

**Comment by nrahul** [comment]
It seems to be fixed in the latest release 0.9.0
